File: OneDrive/Documents/PhaseGateOne/Java/CheckOutApp.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Date;
import java.text.SimpleDateFormat;
public class CheckOutApp {

	public static double purchaseTotal(ArrayList<Double> productPrice, ArrayList<Integer> productQuantity){

	double totalProductAmount = 0;

	for (int index = 0; index < productPrice.size(); index++){
		totalProductAmount +=  (productPrice.get(index) * productQuantity.get(index));
		}

	return totalProductAmount;
	}


	public static double[] productTotal(ArrayList<Double> productPrice, ArrayList<Integer> productQuantity){

	double[] totalProductPrice = new double[productPrice.size()];

	for (int index = 0; index < productPrice.size(); index++){
		totalProductPrice[index] =  (productPrice.get(index) * productQuantity.get(index));
		}

	return totalProductPrice;
	}

	public static double valueAddedTax(ArrayList<Double> productPrice, ArrayList<Integer> productQuantity){

	double total = 0;
	for (int index = 0; index < productPrice.size(); index++){
		total += (productPrice.get(index) * productQuantity.get(index));
		}
	double taxPercentage = 7.5;
	final double PERCENTAGE = 100;
	double vat = (total / PERCENTAGE) * taxPercentage;
	
	return vat;
	}

	public static double purchaseDiscount(ArrayList<Double> productPrice, ArrayList<Integer> productQuantity){

	double totalAmount = 0;
	for (int index = 0; index < productPrice.size(); index++){
		totalAmount += (productPrice.get(index) * productQuantity.get(index));
		}
	double discountPercentage = 3.8;
	final double PERCENTAGE = 100;
	double discount = (totalAmount / PERCENTAGE) * discountPercentage;
	
	return discount;
	}

	public static double billTotal(ArrayList<Double> productPrice, ArrayList<Integer> productQuantity){
	double total = 0;
		for (int index = 0; index < productPrice.size(); index++){
		total += (productPrice.get(index) * productQuantity.get(index));
		}
	double taxPercentage = 7.5;
	final double PERCENTAGE = 100;
	double vat = (total / PERCENTAGE) * taxPercentage;

	double totalToPay = total + vat;

	return totalToPay;
	}

	public static double balance(ArrayList<Double> productPrice, ArrayList<Integer> productQuantity, double amountPaid){

	double totalPurchase = 0;
		for (int index = 0; index < productPrice.size(); index++){
		totalPurchase += (productPrice.get(index) * productQuantity.get(index));
		}

	double taxPercentage = 7.5;
	final double PERCENTAGE = 100;
	double vat = (totalPurchase / PERCENTAGE) * taxPercentage;

		double customersBalance = (amountPaid - (totalPurchase + vat));
	return customersBalance;
	}
	
	public static String storeAddress(){

		String StoreDetails = """
	SEMICOLON STORES
	MAIN BRANCH
	LOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS.
	TEL: 03293828343
	""";

	return StoreDetails;
	}

	public static String currentDateAndTime(){

	Date currentDate = new Date();

	SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");

	SimpleDateFormat timeFormat = new SimpleDateFormat("hh:mm:ss aa");

	String dateForNow = dateFormat.format(currentDate);

	String timForNow = timeFormat.format(currentDate);

	return dateForNow + " " + timForNow;
	}


	public static void main(String[] args){

	Scanner input = new Scanner(System.in);
	
	ArrayList<String> purchasedproducts = new ArrayList<>();
	ArrayList<Double> productPrice = new ArrayList<>();
	ArrayList<Integer> productQuantity = new ArrayList<>();

	
	System.out.println("Enter customer's name: ");
	String customersName = input.next(); 	

	boolean purchase = true;
	while (purchase == true){
	
	System.out.print("Enter name of product: ");
	String goods = input.next();  
	purchasedproducts.add(goods);

	System.out.print("Enter price of product: ");
	double price = input.nextDouble();  
	productPrice.add(price);

	System.out.print("How many did customer buy? : ");
	int quantity = input.nextInt(); 
	productQuantity.add(quantity);

	System.out.println("Would you like to continue. Press \"1\" > Yes or \"0\" > No: ");
	String usersChoice = input.next();
	
		switch (usersChoice){
			case "1": purchase = true; break;
						
			case "0": purchase = false; break;

			default : purchase = false;  break;
			}
	}

	System.out.print("Enter cashier's name: ");	
	String cashiersName = input.next(); 

	String dashes = "";
	for (int dash = 0; dash < 56; dash++){
		dashes += "=";
	}
	
		System.out.println("\n" + dashes);
		System.out.printf("%sDate: %s%n", storeAddress(), currentDateAndTime());
		System.out.println("Cashier: " + cashiersName);
		System.out.println("Customers Name: " +customersName);
		System.out.println(dashes);
		System.out.printf("%15s %8s  %11s %15s%n", "ITEM", "QTY", "PRICE", "TOTAL(NGN)");
		System.out.println(dashes);

		double[] productPriceTotal = productTotal(productPrice, productQuantity);

		for (int count = 0; count < productPrice.size(); count++ ){
		System.out.printf("%15s \t%d \t%.2f \t  %.2f%n", purchasedproducts.get(count), productQuantity.get(count), productPrice.get(count), productPriceTotal[count]);
		}

		System.out.println(dashes);
		System.out.printf(" \t Sub-Total: %.2f%n", purchaseTotal(productPrice, productQuantity));

		System.out.printf(" \t Discount:  %.2f%n", purchaseDiscount(productPrice, productQuantity));
		System.out.printf(" \t VAT@ 7.5%%:  %.2f%n%s%n", valueAddedTax(productPrice, productQuantity), dashes);

		System.out.printf(" \t Bill Total: %.2f%n", billTotal(productPrice, productQuantity));

		System.out.println(dashes);
		System.out.printf(" \t THIS IS NOT A RECEIPT, KINDLY PAY %.2f%n %s%n", billTotal(productPrice, productQuantity), dashes);
		
		System.out.println("");

			System.out.println("How much did the customer pay? ");
			double amountPaid = input.nextDouble(); 

		System.out.println("\n" + dashes);
		System.out.printf("%sDate: %s%n", storeAddress(), currentDateAndTime());

		System.out.printf("Cashier: %s%nCustomers Name:  %s%n%s%n", cashiersName, customersName, dashes);

		System.out.printf("%15s %8s  %11s %15s%n", "ITEM", "QTY", "PRICE", "TOTAL(NGN)");
		System.out.println(dashes);

			for (int count = 0; count < productPrice.size(); count++ ){
			System.out.printf("%15s \t%d\t%.2f \t  %.2f%n", purchasedproducts.get(count), productQuantity.get(count), productPrice.get(count), productPriceTotal[count]);
			}

		double balance = balance(productPrice, productQuantity, amountPaid);

		System.out.println(dashes);
		System.out.printf(" \t Sub-Total: %.2f%n", purchaseTotal(productPrice, productQuantity));
		System.out.printf(" \t Discount:  %.2f%n", purchaseDiscount(productPrice, productQuantity));
		System.out.printf(" \t VAT@ 7.5%% : %.2f%n %s%n", valueAddedTax(productPrice, productQuantity), dashes);
		

		System.out.printf(" \t Bill Total:  \t %.2f%n", billTotal(productPrice, productQuantity));
		System.out.printf("\t Amount Paid: \t %.2f%n", amountPaid);
		if (balance < 0){
			System.out.printf("\t Balance:   \t %.2f%n \t You still owe %.2f. Please, pay up%n", balance, (Math.abs(balance)));
		} else {
			System.out.printf("\t Balance:   \t %.2f%n", balance);
			}
		System.out.println(dashes);
		System.out.println("  \t  THANKS FOR YOUR PATRONAGE");
		System.out.println(dashes);

		}	

}

File: OneDrive/Documents/PhaseGateOne/Python/check_out_app.py

def sub_total(price_of_product, quantity_of_product):
	total_amount = 0
	for index in range(len(price_of_product)):
		total_amount += price_of_product[index] * quantity_of_product[index]

	return total_amount;


def value_added_tax(price_of_product, quantity_of_product):
	#To calculate the total tax amount to be paid.
	total = 0
	for index in range(len(price_of_product)):
		total += price_of_product[index] * quantity_of_product[index]

	PERCENTAGE =100
	tax_percentage = 7.5

	tax_amount = (total / PERCENTAGE) * tax_percentage;
	return tax_amount;

def payment_amount(price_of_product, quantity_of_product):
	#To determine the total payment amount by customer
	
	total_amount = 0
	for index in range(len(price_of_product)):
		total_amount += price_of_product[index] * quantity_of_product[index]

	PERCENTAGE =100
	discount_percentage = 3.8
	tax_percentage = 7.5

	discount_amount = (total_amount / PERCENTAGE) * discount_percentage;
	tax_amount = (total_amount / PERCENTAGE) * tax_percentage;

	amount_to_pay = (tax_amount + total_amount) - discount_amount
	return amount_to_pay;

def purchase_discount(price_of_product, quantity_of_product):
	#To calculate discount for purchase amount.
	totalAmount = 0
	for index in range(len(price_of_product)):
		totalAmount += price_of_product[index] * quantity_of_product[index]

	PERCENTAGE =100
	discount_percentage = 3.8
	
	discount_amount = (totalAmount / PERCENTAGE) * discount_percentage;
	return discount_amount;

def customers_balance(bill, amount_paid):

	balance = (amount_paid - bill)

	return balance;

def header():
	headings = "	ITEM \t QTY \t PRICE \t TOTAL"
	
	return headings

def lines():
	#Printing lines in receipt
	line = "=" * 55
	
	return line;


def store_details():
	#To hold store details
	StoreDetails = """SEMICOLON STORES 
MAIN BRANCH
LOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS.
TEL: 03293828343
"""
	return StoreDetails;


products = []
price_of_product = []
quantity_of_product = []	

customers_name = str(input("Enter customer's name: "))

trade = True
while trade == True:
	product = input('\nEnter name of product: ')
	products.append(product )

	price = float(input("\nEnter price of product: "))
	price_of_product.append(price)

	product_count = int(input('\nHow many did the customer buy? : '))
	quantity_of_product.append(product_count)
	
	users_choice = int(input('\nWould you like to add more? \nPress \"1\" > Yes or \"0\" > No: '))

	if users_choice == 1:
		trade = True
	else:
		trade = False




cashiers_name = str(input("\n Enter cashier's name: "))
bill = float(payment_amount(price_of_product, quantity_of_product))

product_total_price = []
for index in range(len(price_of_product)):
	product_total_price.append(price_of_product[index] * quantity_of_product[index])


print(lines())
print(store_details())

print("Customer's name: ",customers_name)
print(f"Cashier's name: {cashiers_name} \n{lines()}")

print(f'{header()} \n{lines()}')

for item in range(len(products)):
	print(f'\t {products[item]}  \t {quantity_of_product[item]:} \t {price_of_product[item]:.2f} \t{product_total_price[item]:.2f}')

print(lines())

print(f" \t Sub-total: {sub_total(price_of_product, quantity_of_product):.2f} \n \t Discount: { purchase_discount(price_of_product, quantity_of_product):.2f}")
print(f" \t VAX @7.5% : {value_added_tax(price_of_product, quantity_of_product):.2f}\n{lines()}") 
print(f"\t Bill Total: {payment_amount(price_of_product, quantity_of_product):.2f} \n{lines()}")
print(f" THIS IS NOT A RECEIPT, KINDLY PAY {payment_amount(price_of_product, quantity_of_product):.2f} \n{lines()}")

amount_paid = float(input('\nHow much did the customer pay: '))

balance = customers_balance(bill, amount_paid)

print(f'{lines()} \n{header()} \n{lines()}')

for item in range(len(products)):
	print(f'\t {products[item]}  \t {quantity_of_product[item]:} \t {price_of_product[item]:.2f} \t{product_total_price[item]:.2f}')

print(lines())

print(f" \t Sub-total: {sub_total(price_of_product, quantity_of_product):.2f} \n \t Discount: { purchase_discount(price_of_product, quantity_of_product):.2f}")
print(f" \t VAX @7.5% : {value_added_tax(price_of_product, quantity_of_product):.2f}\n{lines()}") 
print(f"\t Bill Total: {payment_amount(price_of_product, quantity_of_product):.2f} \n  \tAmount Paid:  {amount_paid:.2f} \n \tBalance: {balance:.2f}\n")

if balance < 0:
	print(f"\tYou still owe {abs(balance):.2f}, Please pay up")

print(f"{lines()}\n \t THANKS FOR YOUR PATRONAGE \n{lines()}")



